### ISSUE TITLE
Model keeps generating the same character (space) after training

### ISSUE THREAD

**Issue by Kai4nIS4n** [KrishM123/transformer.cpp#6]
**Description:**
Hi, I trained the model following the provided configuration, and the loss converged around **3.3–3.5**, which seems reasonable.
However, during inference, the model keeps generating **the same character repeatedly (a space)** instead of producing meaningful text.

I would like to confirm whether this behavior is expected or if it might indicate a problem in my setup.

**Training configuration:**

```
inference_mode = false
load_existing_weights = false
weights_filename = transformer_weights.bin
data_filename = ../data/tiny_shakespeare.txt

embed_dim = 256
max_sequence_length = 100
num_layers = 32
num_heads = 16
ff_hidden_dim = 1024
dropout_rate = 0.1
pad_token_id = 0.0

learning_rate = 0.0005
num_epochs = 100
batch_size = 128
input_seq_length = 10
decoder_seq_length = 10

max_generate_length = 100
initial_prompt = ROMEO:

num_threads = 32
 
```

**Question:**
Is this expected for the current configuration, or could this be caused by something like improper tokenization, padding setup, or missing normalization during inference?

**Comment by Kai4nIS4n** [comment]
> Hi, are you running from Windows? If so, try this branch:
> 
> [#5](https://github.com/KrishM123/transformer.cpp/pull/5)

Hi, thanks for your reply!
I’m actually running this on Ubuntu 22.04, not Windows.

**Comment by KrishM123** [comment]
The following worked for me when I first built this, however, your configuration is correct and should work. I'll look into this in the coming days. My exe would not work on your machine

```
inference_mode = false
load_existing_weights = yes
weights_filename = transformer_weights.bin
data_filename = ../data/tiny_shakespeare.txt

embed_dim = 1024
max_sequence_length = 500
num_layers = 64
num_heads = 32
ff_hidden_dim = 1024
dropout_rate = 0.1
pad_token_id = 0.0

learning_rate = 0.0005
num_epochs = 1000
batch_size = 128
input_seq_length = 100
decoder_seq_length = 100

max_generate_length = 100
initial_prompt = ROMEO:

num_threads = 32
```

**Comment by Kai4nIS4n** [comment]
> The following worked for me when I first built this, however, your configuration is correct and should work. I'll look into this in the coming days. My exe would not work on your machine
> 
> ```
> inference_mode = false
> load_existing_weights = yes
> weights_filename = transformer_weights.bin
> data_filename = ../data/tiny_shakespeare.txt
> 
> embed_dim = 1024
> max_sequence_length = 500
> num_layers = 64
> num_heads = 32
> ff_hidden_dim = 1024
> dropout_rate = 0.1
> pad_token_id = 0.0
> 
> learning_rate = 0.0005
> num_epochs = 1000
> batch_size = 128
> input_seq_length = 100
> decoder_seq_length = 100
> 
> max_generate_length = 100
> initial_prompt = ROMEO:
> 
> num_threads = 32
> ```

thank you very much！

**Comment by KrishM123** [comment]
Just replaced argmax with softmax in main. Fix works on my end. Pull the latest changes and lmk

**Comment by Kai4nIS4n** [comment]
> Just replaced argmax with softmax in main. Fix works on my end. Pull the latest changes and lmk

thank you very much！i will try it
